Limit number of evaluation samples processed during training

soghomon-b opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

closes #31561 


## Before submitting
- [no ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [yes ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [yes ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
closes #31561 
- [ no] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
There seems to be no documentation for TrainingArgument. I added a line in the training_args.py to explain the new argument. 
- [yes ] Did you write any new necessary tests?
test_number_of_eval_samples_set and test_number_of_eval_samples_unset in tests/trainer/test_trainer.py


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker
- vision models: @amyeroberts, @qubvel
- speech models: @eustlb
- graph models: @clefourrier

Library:

- flax: @gante and @Rocketknight1
- generate: @zucchini-nlp (visual-language models) or @gante (all others)
- pipelines: @Rocketknight1
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @muellerzr and @SunMarc
- chat templates: @Rocketknight1

Integrations:

- deepspeed: HF Trainer/Accelerate: @muellerzr
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc @MekkCyber

Documentation: @stevhliu

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @Rocketknight1
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

soghomon-b committed:
initial commit

eljandoubi committed:
fix Gemma3 Config (#36893)

* fix Gemma3 Config

* fix config in modular gemm3

zucchini-nlp committed:
Fix torch version guard at import (#36907)

fix

JustinTong0323 committed:
[Fix] Add `original_max_position_embeddings` to YARN rope_scaling optional keys (#36877)

[fix] Update optional keys in _validate_yarn_parameters to include original_max_position_embeddings

dvrogozh committed:
tests: fix asyncio.wait() usage for python>=3.11 (#36898)

tests: fix asyncio.wait() usage for python>=3.7

Passing coroutings directly to `asyncio.wait()` is deprecated since
python 3.8 and removed starting from python 3.11. Instead, it's required
to explicitly wrap coroutine in the task with `asyncio.create_task()` which
first appeared in python 3.7.

We step into this issue running the following Transformers tests on a
system with python 3.11 or later (for example, Ubuntu 24.04 has python 3.12):

* `tests/trainer/test_trainer_distributed.py`
* `tests/extended/test_trainer_ext.py`

The error will be:
```
src/transformers/testing_utils.py:2380: in execute_subprocess_async
    result = loop.run_until_complete(
/usr/lib/python3.12/asyncio/base_events.py:687: in run_until_complete
    return future.result()
src/transformers/testing_utils.py:2368: in _stream_subprocess
    await asyncio.wait(
...
E           TypeError: Passing coroutines is forbidden, use tasks explicitly.

```

See: https://docs.python.org/3.10/library/asyncio-task.html#asyncio.wait
See: https://docs.python.org/3.10/library/asyncio-task.html#asyncio.wait
See: https://docs.python.org/3.7/library/asyncio-task.html#asyncio.create_task

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

zucchini-nlp committed:
[chameleon] fix num image token check (#36918)

* [chameleon] fix num image token check

* embed after merging image token

* skip this also

* mistral require_read_token

MekkCyber committed:
Fix Compressed tensors to_dict_diff (#36922)

fix

ydshieh committed:
Use another repo. for Mistral3 processor testing (#36925)

* fix

* fix

* fix

* fix

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

Rocketknight1 commented:
cc @muellerzr @SunMarc

omahs committed:
Fix typos (#36910)

* fix typos

* fix typos

* fix typos

* fix typos

ethanknights committed:
Update `trainer_pt_utils.py` docstrings for consistency (#36912)

* Update trainer_pt_utils.py

* update docstrings trainer_pt_utils.py for consistency

* Update src/transformers/trainer_pt_utils.py

---------

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

cyyever committed:
[2/N] Use pyupgrade --py39-plus to improve code (#36857)

Use pyupgrade --py39-plus to improve code

qubvel committed:
Fix pytorch defomr attn path (#36923)

* Fix pytorch path for DeformableAttention

* Apply for GroundingDino

ydshieh committed:
More precise comment (#36935)

* fix

* fix

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

capemox committed:
Added support for seed in `DataCollatorForWholeWordMask` (#36903)

* Added support for seed in `DataCollatorForWholeWordMask`, and also wrote tests.

Also fixed bugs where the code hardcoded values for mask replacement probability and random replacement probability, instead of using the values passed by the user.

* formatting issues

* Used better way to generate seed in TF. Made tests more consistent.

yonigozlan committed:
Fix processor kwargs qwen2 vl (#36890)

* Fix qwen2_vl and qwen2_5_vl processors cutom images kwargs

* change version warning

MekkCyber committed:
Disallow Offload to disk for gguf files (#36933)

update

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b commented:
Hi 
code is ready and tests are ready to be run. However, as explained above I am unable to run them locally

soghomon-b commented:
Hello, 
Here is a summary of my implementation: 

- Added a new training argument named max_eval_samples, which controls the maximum number of evaluation samples batch to be used during training.
- Ex: if max_eval_samples == 1 and eval_batch_size == 4, only 4 samples will be processed when evaluating model output during training. However, if you call trainer.evaluate() all samples will be used. 
- I wrote one test and the test passed locally. I am not sure why the tests aren't running on CircleCI. 

Happy to do more reviews if needed.

zucchini-nlp committed:
Deprecate #36741 and map Causal to Conditional (#36917)

* deprecate the prev fix

* reword warning and update docs

* reword warning

* tests

* dont bloat `get_text_config()`

Cyrilvallez committed:
Add Phi4 multimodal (#36939)

* raw start

* update

* update

* add to imports

* update

* up

* simplify configs

* clean configs

* style

* typos

* Update convert_phi4_multimodal_weights_to_hf.py

* Update convert_phi4_multimodal_weights_to_hf.py

* fix

* up

* up

* up

* Update convert_phi4_multimodal_weights_to_hf.py

* Update convert_phi4_multimodal_weights_to_hf.py

* up

* up

* up

* Update feature_extraction_phi4_multimodal.py

* up

* up

* up

* up

* up

* simplify configs

* typo

* cut code

* typo

* typo

* typo

* re

* typo

* up

* up

* up

* add tests

* fix

* fix

* Update test_modeling_phi4_multimodal.py

* up

* Update test_modeling_phi4_multimodal.py

* doc

* fix

* up

* up

* up

* up

* up

* up

* simplify

* up

* simplify

* config docstrings

* cleanup

* clean

* typo

* typo

* fix

* Update phi4_multimodal.md

* fix

* fix

* Update test_modeling_phi4_multimodal.py

* update

* simplify reshapes and permutes

* up

* simplify special tokens

* simplify processor a lot

* Update processing_phi4_multimodal.py

* Update processing_phi4_multimodal.py

* switch to fast processor

* image processor

* Update image_processing_phi4_multimodal_fast.py

* add lora extraction to converter

* Update convert_phi4_multimodal_weights_to_hf.py

* Update __init__.py

* add AudioInput type in audio_utils

* rewrite feature_extraction: support torch batched FFT

* input_audio_embeds -> audio_input_features, input_image_embeds -> image_pixel_values

* test update

* not mono channel warning update

* remove auto maps from processor

* kargs dispatch in processor

* simplify kwargs dispatch

* simplify merging

* remove default sampling rate

* style

* Update test_modeling_phi4_multimodal.py

* update doc

* doc

* torch only feature extractor

* make fake tokens adjustable

* Update feature_extraction_phi4_multimodal.py

* fix

* Update processing_phi4_multimodal.py

* simplify mask

* last touch

* fix copies

* style

* Update audio_utils.py

* style

* Update feature_extraction_phi4_multimodal.py

* Update __init__.py

* docstrings

* copies

* fix all checks

* back to fix-copies

* trigger CIs

* Update feature_extraction_phi4_multimodal.py

* improve tests with multimodal inputs

* trigger CIs

---------

Co-authored-by: Eustache Le Bihan <eulebihan@gmail.com>

MekkCyber committed:
Fixing _pre_quantization_dtype when torch_dtype is None (#36930)

fix

guangy10 committed:
Export for Phi4-mini (#36780)

* Export for Phi4-mini

* Update tests/models/phi3/test_modeling_phi3.py

---------

Co-authored-by: Guang Yang <guangyang@fb.com>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

Commit:
fix typos in the tests directory (#36932)

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: fix typos in test codes

* chore: format codes

zucchini-nlp committed:
Support `return_tensors` in audio chat templates (#34601)

* add audio chat templates

* update

* update

* nit

* green ci

* we dont care about the order anymore

* clean up after rebase

* overriden tests rename

* rename shieldgemma also

* one more rename

* require_read_token

* removde images/videos

* retrigger CI flaky

SunMarc committed:
Fix cuda index issue in cache allocator (#36937)

fix

gante committed:
[Utils] torch version checks optionally accept dev versions (#36847)

SunMarc reviewed (commented):
Thanks for the PR. Can you fix your linter. There are too many changes in this PR

ydshieh committed:
Update ruff to `0.11.2` (#36962)

* update

* update

* update

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Update after #36962 (#36965)

update

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

soghomon-b commented:
> Thanks for the PR. Can you fix your linter. There are too many changes in this PR

Hi 
When I run the formatting these changes happen. If I don't run them the tests from circlci don't pass. Should I not run them?

zhanluxianshen committed:
Change GPUS to GPUs (#36945)

Signed-off-by: zhanluxianshen <zhanluxianshen@163.com>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

NargiT committed:
typo fixed in README_fr.md (#36951)

Sai-Suraj-27 committed:
Updated docker files to use `uv` for installing packages (#36957)

* Updated docker files to use uv pip install as uv is blazingly fast.

* Removed -y flag for uv pip uninstall.

* Passed --no-build-isolation flag

---------

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

ydshieh committed:
update examples after ruff being updated (#36972)

* update

* update

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

cyyever committed:
Remove extra tensor clone in PyTorch code (#36748)

* Use detach().clone()

* Eliminate continuous()

* Merge clone and other calls with to

* Merge clone and other calls with to

stevhliu committed:
[docs] Fix image link (#36869)

* fix image link

* fix

* update

* fix

cyyever committed:
Add ruff target-version (#36971)

ydshieh committed:
update bot comment again (#36974)

update

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

yonigozlan committed:
🚨Deprecate legacy argument for image-text-to-text models and adopt new behavior by default (#36307)

* deprecate legacy argument and adopt new behavior by default

* revert back modification git

cyyever committed:
Fix tensor dtype mismatch (#36985)

* Fix tensor dtype mismatch

* update

* update

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
byebye CircleCI TF jobs (#36998)

* byebye tf jobs

* byebye tf jobs

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

cyyever committed:
Use torch.expm1 (#36995)

ydshieh committed:
Install `networkx==3.2.1` manually in some CircleCI jobs after #36957 (#37000)

fix

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

cyyever committed:
Fix Optional type annotation (#36841)

* Fix annotation

* Update src/transformers/generation/candidate_generator.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Update src/transformers/generation/utils.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Update src/transformers/generation/utils.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

---------

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

ivarflakstad committed:
Fix get_device_properties (#36997)

Fix remove remnant self from get_device_properties

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

Cyrilvallez committed:
Allow easy registration of custom attention functions (#36889)

* Update modeling_utils.py

* style

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* add to init

* Update modeling_utils.py

* style

* update

* Update modeling_utils.py

* Update modeling_utils.py

* style

* Add some doc

* Update _toctree.yml

* readd it for tgi/vllm compat

* CIs

* CIs

anadon committed:
Fix removing "cpu" from frozenset in bitsandbytes.py to allow better ROCm support. (#36975)

* Fix removing "cpu" from frozenset in bitsandbytes.py to allow better ROCm support.

Related to https://github.com/bitsandbytes-foundation/bitsandbytes/issues/1573 and https://github.com/huggingface/transformers/issues/36949 , this resolves a bug in allowing ROCm/HIP support in bitsandbytes.

* Related to bitsandbytes-foundation/bitsandbytes#1573 and huggingface#36949 , this resolves a bug in the biteandbytes integration, allowing ROCm/HIP support in bitsandbytes.

---------

Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

MekkCyber committed:
Fix device_map check for ggml files (#37003)

fix

soghomon-b committed:
added test

soghomon-b committed:
fixed small issue

soghomon-b commented:
Hello 
This should be clearer now, I will run the code styling scripts after you approve it.

SunMarc committed:
Log the correct learning rate (#36973)

* fix learning rate log

* fix lr log

* add lr

soghomon-b committed:
added one more test

Commit:
fix typos in the code comments and error messages (#36993)

* chore: enhance code comments

* chore: enhance code comments

* chore: enhance code comments

* chore: enhance code comments

* chore: enhance code comments

* chore: enhance code comments

* chore: enhance code comments

cyyever committed:
Remove deprecated training arguments (#36946)

* Remove deprecated training arguments

* More fixes

* More fixes

* More fixes

stevhliu committed:
[docs] Attention mask image (#36970)

add image

yao-matrix committed:
fix transformers_cli import relative path issue (#36989)

* fix transformers_cli relative import path issue

Signed-off-by: Yao, Matrix <matrix.yao@intel.com>

* fix style

Signed-off-by: Yao, Matrix <matrix.yao@intel.com>

---------

Signed-off-by: Yao, Matrix <matrix.yao@intel.com>
Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

bakrianoo committed:
Support QuestionAnswering Module for ModernBert based models. (#35566)

* push ModernBertForQuestionAnswering

* update ModernBertForQuestionAnswering

* update __init__ loading

* set imports for ModernBertForQuestionAnswering

* update ModernBertForQuestionAnswering

* remove debugging logs

* update init_weights method

* remove custom initialization for ModernBertForQuestionAnswering

* apply make fix-copies

* apply make style

* apply make fix-copies

* append ModernBertForQuestionAnswering to the pipeline supported models

* remove unused file

* remove invalid autoload value

* update en/model_doc/modernbert.md

* apply make fixup command

* make fixup

* Update dummies

* update usage tips for ModernBertForQuestionAnswering

* update usage tips for ModernBertForQuestionAnswering

* add init

* add lint

* add consistency

* update init test

* change text to trigger stuck text

* use self.loss_function instead of custom loss

By @Cyrilvallez

Co-authored-by: Cyril Vallez <cyril.vallez@gmail.com>

* Update modeling_modernbert.py

make comparable commit to even it out

* Match whitespace

* whitespace

---------

Co-authored-by: Matt <rocketknight1@gmail.com>
Co-authored-by: Orion Weller <wellerorion@gmail.com>
Co-authored-by: Orion Weller <31665361+orionw@users.noreply.github.com>
Co-authored-by: Cyril Vallez <cyril.vallez@gmail.com>

mgoin committed:
Fix PixtralProcessor patch_size when spatial_merge_size is used (#37019)

kylesayrs committed:
[Modeling] Load FP8 safetensors such as DeepSeek (#36828)

support loading fp8

Signed-off-by: Kyle Sayers <kylesayrs@gmail.com>
Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

ydshieh committed:
Mark 2 tests as flaky for now (#37038)

* fix

* fix

* fix

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

hiyouga committed:
remove redundant code in trainer (#36994)

* Update optimization.py

* Update optimization.py

MekkCyber committed:
Skip FP8 linear tests For device capability < 9.0(#37008)

* skip fp8 linear

* add capability check

* format

keetrap committed:
Add Distill Any Depth (#36614)

* Added conversion Script

* Update src/transformers/models/depth_anything/convert_distill_any_depth_to_hf.py

Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

* Updated Conversion Script

* Update src/transformers/models/depth_anything/convert_distill_any_depth_to_hf.py

Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

---------

Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

jiqing-feng committed:
fix pegasus init weights and other copied models (#36844)

* fix pegasus init weights

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix the rest of models

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix test

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix informer init

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* init weight before checking

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix roformer tests

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix roformer tests

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

---------

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

n0gu-furiosa committed:
Optimize `to_py_obj` for python-native numeric lists and scalars (#36885)

* Optimize to_py_obj for python-native numeric lists and scalars

* Fix bug that tuple is not converted to list

* Try np.array for more robust type checking

* Apply review and add tests for to_py_obj

qubvel committed:
Fixup for distill_any_depth conversion script (#37043)

* Fixup

* trigger

zucchini-nlp committed:
[chat templates} support loading audio from video (#36955)

* add audio from video

* typos

* delete print

* comments

eustlb committed:
[audio utils] fix fft_bin_width computation (#36603)

* fix fft_bin_width computation

* update docstring + enforce correct params

* update test with correct value

* udpate test

* update feature extractors for concerned models

* update

* make

* udpate docstring

* udpate docstring

gante committed:
[generate, cache] handle more complex device maps (#37014)

zhanluxianshen committed:
clean pipeline question_answering. (#36986)

Signed-off-by: zhanluxianshen <zhanluxianshen@163.com>

cyyever committed:
Avoid unnecessary device operations in loss computing (#36950)

* Avoid unnecessary tensor copy in loss computing

* Add type

cyyever committed:
Fix typing for None valued variables (#37004)

Fix typing for None-able variables

cyyever committed:
Set weights_only in torch.load (#36991)

premmurugan229 committed:
Replace default split function with jnp.split() in flax models (#37001)

Replace split with jnp's split function for flax models (#36854)

cyyever committed:
Remove deprecated batch_size parameter (#37007)

finnoh committed:
fixed typo (#37036)

Wheest committed:
fix: Fully remove legacy cache from Llama (#36958)

* bug: fully remove legacy cache from Llama

* bug: fix CI issues

* bug: update jetmoe model

* bug: apply =check_modular_conversion.py= fix

* bug: apply make fix-copies

* bug: fix ruff

* PR suggestions

* Remove trailing commas in auto-gen files

* Trivial new line removal

ManuelFay committed:
Fix SDPA implementation in Qwen2-VL (issues with torch==2.6.0) (#36891)

* fix sdpa implementation

* ruff

* also modify 2_5 for consistency

jp1924 committed:
fix: AttributeError: 'LlavaProcessor' object has no attribute 'image_token_id' (#37026)

* Add image_token_id and video_token_id handling in Llava processors

* fix: image to video

* fix: correct image and video token ID handling in Llava processors

* fix: improve image and video token ID handling in Llava processors

ydshieh committed:
Use `lru_cache` for tokenization tests (#36818)

* fix

* fix

* fix

* fix

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

zhanluxianshen committed:
Fix some typos about benchmark scripts. (#37027)

Signed-off-by: zhanluxianshen <zhanluxianshen@163.com>

cyyever committed:
Change deprecated PT functions (#37041)

Change deprecated functions

zucchini-nlp committed:
[blip-2] Fix dtype mismatch when keep in fp32  (#37068)

* fix fp32 BLIP2

* no need to reorder that

* check for `Noneness` as well before casting dtype

bzantium committed:
[WIP] add deepseek-v3 (#35926)

* init commit

* style

* take comments into account

* add deepseekv3 modeling

* remove redundant code

* apply make style

* apply fix-copies

* make format

* add init files

* rename deepseekv3 into deepseek_v3 based on its model_type

* rename deepseekv3 into deepseek_v3 based on its model_type

* deepseek-v3 not deepseek_v3

* set model_type as deepseek_v3

* use default docs

* apply make

* fill type and docstring

* add rope_config_validation

* use custom DeepseekV3MLP

* hold code only for checkpoints congifuration; remove redundant

* revise rope yarn for DeepSeek variation

* rename DeepSeek-V3

* some refactoring

* revise load_hook to work properly; make moe func trainable; use llama instead of mixtral

* fix attention forward

* use -1 for not-changing dim when to use exapnd

* refactor DeepseekV3TopkRouter

* use reshape_for_rope instead of load_hook; revise attention forward for TP; rename q_head_dim with qk_head_dim

* register pre_hook and hook both

* make style

* use n_shared_experts

* Update src/transformers/models/deepseek_v3/configuration_deepseek_v3.py

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* add test file

* update modeling_file according to modular file

* make style

* add mapping for DeepseekV3ForSequenceClassification

* remove aux_loss_alpha

* add deepseek_v3 for perf

* add deepseek_v3

* rename test as deepseekv3

* use tiny-deepseek-v3

* remove DeepseekV3ForSequenceClassification

* cache before padding

* remote output_router_logits

* Revert "remote output_router_logits"

This reverts commit f264f800d04950390db8413b9efb24cef8186330.

* remove output_router_logits

* make e_score_correction_bias as buffer

* skip tests not compatible

* make style

* make e_score_correction_bias as buffer

* use rope_interleave instead of load_hook

* skip tests not compatible with MLA

* add doc for rope_interleave

* fix typo

* remove torch.no_grad for selecting topk

* fix post merge issue

* mrege with main and simplify

* nits

* final

* small fixes

* fix

* support TP better

* stash

* changes currently requires

* remove synch

* more fixes for TP

* temp fix for TP : some attention layers's FP8 scales are too small + shared is local colwise and anything is local if FP8 because weights are used

* updates to have generation work!

* push most of the changes

* reorder functions + call for contributions!

* update readme

* nits

* update

* ruff was updated on main

* merge with main and fix copies

* revert unrelated changes

* route all tokens to all experts when testing to avoid no gradient iddues

* finish fixing all tests

* fixup

* nit

* clean config

* last readme changes

* nit

* do cnit

* typo

* last nit

* one more one more

---------

Co-authored-by: Arthur Zucker <arthur.zucker@gmail.com>
Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>
Co-authored-by: arthur@huggingface.co <arthur@ip-26-0-165-131.ec2.internal>

ydshieh committed:
fix tied weigths issue  (#37031)

* fix

* comment

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

muellerzr committed:
Update w/ new account (#37084)

* Update w/ new account

* DS

Cyrilvallez committed:
Fix state_dict map location when quantized (#37086)

* Update modeling_utils.py

* Update modeling_utils.py

Cyrilvallez committed:
Fix AttentionInterface following feedback (#37010)

* up

* typo

* update doc

* Update attention_interface.md

zhanluxianshen committed:
fixed typo. (#37057)

Signed-off-by: zhanluxianshen <zhanluxianshen@163.com>
Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

gante committed:
[generate] beam search -- fix output cropping (#37080)

* handle jagged beams

* better comment

* bart -- beam search tests print special tokens

* more bart test updates

* more tests!

* better comment

gante committed:
[Cache] rename dtype attribute 🚨 🚨  (#37044)

* yoink

* same pattern in all cache

ydshieh committed:
Kenlm (#37091)

* kenlm

* kenlm

* kenlm

* kenlm

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

samrae7 commented on src/transformers/training_args.py:
```diff
@@ -309,6 +309,9 @@ class TrainingArguments:
             If set to a positive number, the total number of training steps to perform. Overrides `num_train_epochs`.
             For a finite dataset, training is reiterated through the dataset (if all data is exhausted) until
             `max_steps` is reached.
+        max_eval_samples (`int`, *optional*, defaults to None)
```
Also, as this is batches and not samples, how about `max_eval_batches`?

soghomon-b replied:
Yes! This is a mistake, thanks for spotting. As for the name, it is also changed!

samrae7 commented on src/transformers/trainer.py:
```diff
@@ -4385,6 +4395,10 @@ def evaluation_loop(
+        
+        #for testing purposes
+        self.observed_num_examples = observed_num_examples
+        self.max_eval_samples = max_eval_samples
```
As far as possible I feel like it's good practice to not add new properties to the class instance only for testing. What do you think? I was thinking we could extend the Trainer class in the test only, like this

```
class MyTrainer(Trainer):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.eval_batches_seen = 0

    def prediction_step(self, model, inputs, prediction_loss_only, ignore_keys=None):
        self.eval_batches_seen += 1
        return super().prediction_step(model, inputs, prediction_loss_only, ignore_keys)
 ```
 
 and then use that class in the test. That way you don't change the actual Trainer class itself. The slight disadvantage would be that you would have to test first the trainer.train(), then reinstantiate and test trainer.evaluate ( as the eval_batches_seen won't start from zero each time) but I think that's ok
 
 ==
 
 If you go ahead with your approach, I feel like the name of the property set on the class should reflect that it is observed_eval_samples (ie. not training examples). I know it's just an ephemeral thing that you're using for testing but it could cause confusion (?)

soghomon-b replied:
Great point too, I was not happy with adding more fields to the object, but I did not know how to test it otherwise. This way of testing is new to me, thanks for bringing it up!

samrae7 reviewed (commented):
Nice work 👏 

Most of my comments are subjective so feel free to ignore and resolve as you see fit. Hopefully one or two are helpful

I am completely new to this project and only reviewed this because I was looking at the issue myself. So you still need a community member to review ( maybe they can weigh-in on the comment about how to 'track samples' seen during the test).

soghomon-b committed:
imporved testing, fixed small documentation error

soghomon-b committed:
removed small unneeded import

soghomon-b committed:
added explanaiton to tests

soghomon-b commented on tests/trainer/test_trainer.py:
```diff
@@ -1293,6 +1293,81 @@ def test_evaluation_with_keys_to_drop(self):
+        self.assertEqual(trainer.max_eval_samples, trainer.observed_num_examples)
+
+
+        trainer.evaluate()
```
Good point, it is not super intuitive, so I added it. Thanks for the suggestion

soghomon-b commented:
Hello @muellerzr @SunMarc 
The PR is ready for a review :)

MinJu-Ha committed:
🌐 [i18n-KO] Translated `qwen2_vl.md` to Korean (#36750)

* fix: manual edits

* fix: resolve suggestions

* Update toctree.yml

bozheng-hit committed:
Adding Qwen3 and Qwen3MoE (#36878)

* Initial commit for Qwen3

* fix and add tests for qwen3 & qwen3_moe

* rename models for tests.

* fix

* fix

* fix and add docs.

* fix model name in docs.

* simplify modular and fix configuration issues

* Fix the red CI: ruff was updated

* revert ruff, version was wrong

* fix qwen3moe.

* fix

* make sure MOE can load

* fix copies

---------

Co-authored-by: Arthur Zucker <arthur.zucker@gmail.com>

yuanwu2017 committed:
Gaudi: Fix the pipeline failed issue with hpu device (#36990)

* Gaudi: fix the issue of is_torch_hpu_available() returns false

Signed-off-by: yuanwu <yuan.wu@intel.com>

* Fix make fixup

Signed-off-by: yuanwu <yuan.wu@intel.com>

* Add comments for the implicit behavior of import

Signed-off-by: yuanwu <yuan.wu@intel.com>

* Update src/transformers/utils/import_utils.py

* Update src/transformers/utils/import_utils.py

---------

Signed-off-by: yuanwu <yuan.wu@intel.com>
Co-authored-by: Ilyas Moutawwakil <57442720+IlyasMoutawwakil@users.noreply.github.com>

efsotr committed:
Support passing flash_attn_kwargs when gradient_checkpointing is enabled (#37037)

* support passing flash_attn_kwargs when gradient_checkpointing is enabled

* make modeling_deepspeek_v3.py consistent with modular_deepseek_v3.py

Qubitium committed:
Fix 4090/ada not detected as having FP8 support (#37067)

fix 4090/ada not detected as having FP8 support

Signed-off-by: Qubitium <qubitium@modelcloud.ai>
Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>
Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

jiqing-feng committed:
enable tp on CPU (#36299)

* enable tp on CPU

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* get rank from cpu

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* update

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* enable TP tests

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix comment

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* em print

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix model id

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix conflict

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix index and add doc

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

---------

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

jiqing-feng committed:
fix whisper re-compile (#36712)

* fix whisper re-compile

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix copy

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix comment

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix copies

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* revert useless changes

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

---------

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>
Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

huismiling committed:
[MLU] Fix FA2 check error, remove deepspeed-mlu deps. (#36159)

* add Cambricon MLUs support

* fix mlu device rng state

* up for quality check

* up mlu to support fp16

* fix mlu device dependency error

* fix mlu device dependency error

* enable mlu device for bf16

* fix mlu device memory tracker

* Cambricon support SDPA and flash_attn

* MLU devices : Checks if `mlu` is available via an `cndev-based` check which won't trigger the drivers and leave mlu

* Fix mlu FA2 check. Remove deepspeed-mlu check. add mlu tests support.

* fix testing errors.

* Merge branch 'hf/main' into main

* fix get_device_count error.

* fix mlu testing utils.

* fix code quality and style.

* switch to @require_torch_multi_accelerator

gau-nernst committed:
Fix Gemma3 embedding scaling (#37109)

fix gemma3 embedding

RobinKa committed:
RWKV: fix mask warning typo (#37114)

rwkv: fix mask warning typo

cyyever committed:
Remove deprecated code (#37059)

* Remove deprecated code

* fix get_loading_attributes

* fix error

* skip test

---------

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>
Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

geetu040 committed:
Create and Expose SamVisionModel as public for better accessibility (#36493)

* move encoder below

* auto modeling

* write SamVisionTester

* fix vision attention shape

* fix SamVisionTest

* minor changes to SamVisionTest

* Revert "fix vision attention shape"

This reverts commit d2a4083ae5704716e33351aed03af8f3cc45f3ae.

* fix attention output shape in new tests

* remove encoder examples

* run modular on got_ocr2

* code formatting

* fix got_ocr2

* ruff fixes

* code quality

* add sam_vision in auto modeling and auto configuration

* remove composite test

* updated index.md

* add TFSamVisionEncoder to __init__

* fix public TFSamVisionEncoder

* remove outdated todo comment

* set test_torch_exportable

Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

* rename: VisionEncoder -> VisionModel

* bring back original SamVisionEncoder

* rename back: VisionEncoderOutput -> VisionModelOutput

* undo changes in SamModelTester

* reuse SamVisionEncoder in SamVisionModel

---------

Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

faaany committed:
[tests] remove cuda-only test marker in `AwqConfigTest`  (#37032)

* enable on xpu

* add xpu support

---------

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

guangy10 committed:
Export T5 (encoder-decoder) to ExecuTorch (#36486)

Co-authored-by: Guang Yang <guangyang@fb.com>

ydshieh committed:
skip (#37141)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

FightingZhen committed:
[Feature] Support using FlashAttention2 on Ascend NPU (#36696)

* [Feature] Support using flash-attention on Ascend NPU

* Fix qwen3 and qwen3_moe moduler conversion mismatch

zucchini-nlp committed:
[qwen3] fix generation tests (#37142)

* do not skip tests

* fix qwen3-moe as well

* fixup

* fixup

Cyrilvallez committed:
Remove low_cpu_mem_usage and _fast_init (#36963)

* Remove low_cpu_mem_usage and _fast_init

* Update deepspeed.py

* Update modeling_utils.py

* remove the first 2 tests everywhere

* Update test_modeling_common.py

* remove what was remaining about fast_init

* fix logic and simplify

* mismatched keys logic update

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* Update modeling_utils.py

* fix 2 models init_weights

* extend to others

* remove grad

* Update modeling_fsmt.py

* init weights in tests

* style

* Update test_modeling_fsmt.py

* more old models

* fix more init_weights

* copies

* fix

* style

* Update modeling_lxmert.py

* fix inits

* more and more

* more

* should finalize

* style

* Update modeling_dinov2_with_registers.py

* fix

* Update modeling_encoder_decoder.py

* fix

* style

* Update modeling_lxmert.py

* post rebase cleanup

* Update modeling_informer.py

* back to start for device

* fix

* add test to detect all failing cases correctly

* Update test_modeling_common.py

* fix

* fix

* sam

* style

* Update modeling_maskformer_swin.py

* CIs

* CIs

* remove test - will add it on separate PR

* fix

* fix

* Update modeling_sam.py

* CIs

* CIs

* CIs

* convnext

* suggestions

* CIs

* fix copies after merge

---------

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

qubvel committed:
Refactor `return_dict` logic to remove complicated if/else paths (#36794)

* SAM

* CLIP

* SigLIP

* GOT-OCR2 (depends on SAM)

* SigLIP2 (depends on SigLIP)

* trigger tests

* Fix SAM

* Fix missed indexing, use named attributes

* Llama

* Aria

* Bamba

* Update llama: missed outputs return type

* (fixup) Aria

* DiffLlama

* Emu3

* Gemma

* Gemma2

* Paligemma

* Fix paligemma

* Gemma3

* GLM

* Helium

* JetMoe

* Jamba

* Mistral

* Mistral

* Mixtral

* Nemotron

* Olmo

* Olmo2

* Persimmon

* Phi

* Phi3

* PhiMoe

* Qwen2

* Qwen2_moe

* StableLM

* Starcoder2

* Add return_dict decorator

* SAM

* Update decorator: compile, export, trace - friendly

* Llama (decorator)

* SAM (decorator)

* Add decorator `can_return_tuple`

* Llama

* Update to decorator

* Update CLIP

* Update decorator to store `_is_top_level_module` in self

* Update decorator to correctly handle compile/export

* Remove is_torchdynamo_compiling constraint, all work fine with self attribute assignment

* Typing

* GPT NeoX

* Fixup

* Fix attribute Granite

* Fix return type mixtral

* Update Gemma3

* Fix Cohere amd Cohere2

* Fixup

* Fix corner case for Phi4, when activation is shared

* (fix-copies) deepseekv3, phi4

* Fixup

* Apply to qwen3/qwen3_moe

* Fix

cyyever committed:
Fix more inefficient PT operations (#37060)

* Fix inefficient operations

* Remove cpu() call

* Reorder detach()

* Reorder detach()

* tolist without detach

* item without detach

* Update src/transformers/models/rag/modeling_rag.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Update tests/models/encodec/test_modeling_encodec.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Use detach().cpu().numpy

* Revert some numpy operations

* More fixes

---------

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

muellerzr reviewed (approved):
Seems handy, nice :)

muellerzr commented:
@soghomon-b could you rebase and fight the CI a bit until 🟢 ? We can merge it then

soghomon-b committed:
fixing checkstyle

soghomon-b committed:
test2

soghomon-b commented:
Hi @muellerzr 
I have been rerunning tests for a long time, some tests failed but I never touched, I reran it and now it says I have no access to run them.

samrae7 reviewed (approved):
Hey. I'm glad my suggestions were helpful. Thanks 👍

yaswanth19 committed:
Fix std initialization in Idefics variants (#37100)

* Nit 😅

* Another one

* fix

* run ci

* revert change

jiqing-feng committed:
add gpt2 test on XPU (#37028)

* add gpt2 test on XPU

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* auto dtype has been fixed

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* convert model to train mode

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

---------

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

jiqing-feng committed:
Fix llava xpu tests. (#37130)

* fix llava 4bit xpu test

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix llava 4bit xpu test

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix format

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

* fix format

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

---------

Signed-off-by: jiqing-feng <jiqing.feng@intel.com>

yao-matrix committed:
enable `test_assisted_decoding_in_different_gpu` test on XPU (#37120)

Signed-off-by: YAO Matrix <matrix.yao@intel.com>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

guangy10 committed:
Use public export API on torch 2.5 and future (#36781)

Co-authored-by: Guang Yang <guangyang@fb.com>
Co-authored-by: Pavel Iakubovskii <qubvel@gmail.com>

Tavish9 committed:
Convert `_VALID_DICT_FIELDS` to class attribute for shared dict parsing in subclasses (#36736)

* make _VALID_DICT_FIELDS as a class attribute

* fix test case about TrainingArguments

IlyasMoutawwakil committed:
Only count num items in batch when needed (#36867)

only count num itels when needed

tugsbayasgalan committed:
Make canine model exportable by removing unncessary complicated logic (#37124)

tomaarsen committed:
[`ModernBERT`] Never save 'reference_compile' config; should be set based on end user (#36305)

* Never save 'reference_compile' config; should be set based on end user

* Reformat (I ran 'make style' from the wrong env)

* Use pop instead of del

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

* Use pop instead of del

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

---------

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

yao-matrix committed:
fix XPU UT error case brough by RNG difference btw XPU and CUDA (#37121)

* fix XPU UT error case brough by RNG difference btw XPU and CUDA

Signed-off-by: YAO Matrix <matrix.yao@intel.com>

* enable tests/models/llama/test_modeling_llama.py::LlamaIntegrationTest::test_model_7b_logits and tests/models/llama/test_modeling_llama.py::LlamaIntegrationTest::test_model_7b_logits_bf16 on xpu

Signed-off-by: YAO Matrix <matrix.yao@intel.com>

* Revert "enable tests/models/llama/test_modeling_llama.py::LlamaIntegrationTest::test_model_7b_logits and tests/models/llama/test_modeling_llama.py::LlamaIntegrationTest::test_model_7b_logits_bf16 on xpu"

This reverts commit 3ef83a4f0204642daa45fda56e8aca1afed24b4f.

---------

Signed-off-by: YAO Matrix <matrix.yao@intel.com>

qubvel committed:
Refactor attention for SigLIP based models (#36981)

* Update Siglip attention implementation

* Update tests for Siglip

* Remove one level of indentation

* Update test to be more specific

* Fixup

* Idefics2

* Idefics3

* Emu3

* SmolVLM

* Phi4 (just init small update)

* Idefics2 (test fix)

* Update siglip2 tests

* Update eager

* trigger

* Clean up

* Transfer inputs to device in test

* Fixing test

* Fixing test

* Revert contiguous

* Remove unused is_flash_attn_2_available

* Move flaky to specific models

Zephyr271828 committed:
Fixes the inconsistency of the optionality of attention_mask (#37153)

* debugging issue 36758

* debugging issue 36758

* debugging issue 36758

* updated attn_mask type specification in _flash_attention_forward

* removed pdb

* added a blank line

* removed indentation

SunMarc committed:
Merge branch 'main' into add-eval-step-limit-31561

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_36916). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

ydshieh committed:
Avoid pipeline test failing related to Hub call (#37170)

* cls

* cls

* cls

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

Cyrilvallez committed:
Fix meta state dict loading with quantizers (#37136)

Update modeling_utils.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

Cyrilvallez committed:
Revert #37031 (#37178)

Update modeling_utils.py

BowenBao committed:
[doc] Fix link for Quark quantization page (#37179)

Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

soghomon-b commented:
@muellerzr @SunMarc 
Hello,
There are two tests failing, both are not related to my changes and Im not sure how to fix them, would appreciate your help.

zucchini-nlp committed:
[chat-template] fix video loading (#37146)

* fix

* add video

* trigger

* push new iamges

* fix tests

* revert

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Skip code `307` in `RequestCounter` (#36953)

fix

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

jerryzh168 committed:
Add device workaround for int4 weight only quantization after API update (#36980)

* merge

* fix import

* format

* reformat

* reformat

---------

Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

xadupre committed:
Fixes DynamicCache export issues due to control flow and inplace modifications (#36652)

* Remove unnecessary masked_fill in deberta models

* Enable some code when exporting but not compiling

* add missing import

* style

* replace if by torch.cond

* style

* use numel

* style

* add unit tests

* style

* change empty value for dynamic cache

* replace != [] by numel()

* fix import issue

* style

ydshieh committed:
Try to avoid/reduce some remaining CI job failures (#37202)

* try

* try

* Update tests/pipelines/test_pipelines_video_classification.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

---------

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

saattrupdan committed:
fix: Add 'image-text-to-text' to `TASK_MAPPING` (#37107)

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

zhanluxianshen committed:
Fix some code annotation typos. (#37102)

Signed-off-by: zhanluxianshen <zhanluxianshen@163.com>

cyyever committed:
Merge tensor operations with device transfer operations (#37097)

* Merge operations with to

Signed-off-by: cyy <cyyever@outlook.com>

* Use dtype

Signed-off-by: cyy <cyyever@outlook.com>

---------

Signed-off-by: cyy <cyyever@outlook.com>

cyyever committed:
[3/N] Use pyupgrade --py39-plus to improve code (#36936)

Use pyupgrade --py39-plus to improve code

Signed-off-by: cyy <cyyever@outlook.com>

cyyever committed:
Add py.typed (#37022)

Rocketknight1 committed:
No more dtype_byte_size() (#37144)

* No more dtype_byte_size()

* Remove function once again

* Fix rebase cruft

* Trigger tests

gante committed:
[Tests] add `min_new_tokens` to prevent flaky length checks (#37175)

SunMarc committed:
Merge branch 'main' into add-eval-step-limit-31561

SunMarc commented on src/transformers/training_args.py:
```diff
@@ -295,6 +295,9 @@ class TrainingArguments:
             `max_steps` is reached.
+        max_eval_batches (`int`, *optional*, defaults to -1)
+            The maximum number of evaluation samples batches to be used per epoch during training only.
+            If not set to any number, the whole evaluation dataset will be used during training.
```
```suggestion
           The whole evaluation dataset will be used during training by default
```

SunMarc commented on tests/trainer/test_trainer.py:
```diff
@@ -1293,6 +1304,76 @@ def test_evaluation_with_keys_to_drop(self):
+            eval_dataset=eval_dataset,
+        )
+
+        # when calling train, it will set limit_max_eval_sample to True and will apply limit
```
```suggestion
        # when calling train, we will limit the eval sample size if max_eval_batches is defined as we have limit_eval_sample_size=True in _evaluate method 
```

SunMarc commented on tests/trainer/test_trainer.py:
```diff
@@ -1293,6 +1304,76 @@ def test_evaluation_with_keys_to_drop(self):
+        # reset number of batches seen
+        trainer.observed_num_batches = 0
+
+        # when calling evaluate directly, it will set limit_max_eval_sample to False and won't apply the limit
```
```suggestion
        # when calling evaluate directly, we won't limit the eval sample size if max_eval_batches is defined as we have limit_eval_sample_size=False 
```

SunMarc commented on tests/trainer/test_trainer.py:
```diff
@@ -1293,6 +1304,76 @@ def test_evaluation_with_keys_to_drop(self):
+            eval_dataset=eval_dataset,
+        )
+
+        # when calling train, it will set limit_max_eval_sample to True and will apply limit
```
```suggestion
        # when calling train, we will limit the eval sample size if max_eval_batches is defined as we have limit_eval_sample_size=True in _evaluate method 
```

SunMarc commented on tests/trainer/test_trainer.py:
```diff
@@ -1293,6 +1304,76 @@ def test_evaluation_with_keys_to_drop(self):
+        # reset number of batches seen
+        trainer.observed_num_batches = 0
+
+        # when calling evaluate directly, it will set limit_max_eval_sample to False and won't apply the limit
```
```suggestion
         # when calling evaluate directly, we won't limit the eval sample size if max_eval_batches is defined as we have limit_eval_sample_size=False 
```

SunMarc commented on src/transformers/trainer.py:
```diff
@@ -3088,7 +3088,8 @@ def _maybe_log_save_evaluate(
         metrics = None
         if self.control.should_evaluate:
-            metrics = self._evaluate(trial, ignore_keys_for_eval)
+            # When calling evaluate directly, it will only limit with max_eval_sample=True
```
A bit unclear. Maybe we can move this comment somewhere else ? Also, here a better version : 
we will only limit the eval sample size if `max_eval_batches` is also set when setting limit_eval_sample_size = True
```suggestion
```

SunMarc reviewed (commented):
LGTM ! Just a few nits

soghomon-b committed:
Update src/transformers/trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update src/transformers/trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update src/transformers/training_args.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update src/transformers/training_args.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
Update tests/trainer/test_trainer.py

Co-authored-by: Marc Sun <57196510+SunMarc@users.noreply.github.com>

soghomon-b committed:
fixed style

soghomon-b committed:
fixed style

soghomon-b commented:
@SunMarc 
It keeps saying "would reformat tests/trainer/test_trainer.py" however, I ran make style.

soghomon-b committed:
ran styling

soghomon-b committed:
ran styling

soghomon-b committed:
ci: retrigger PR slow CI sha detection

soghomon-b committed:
ci: retrigger PR slow CI sha detection

soghomon-b committed:
ci: retrigger PR slow CI sha detection

soghomon-b committed:
Merge branch 'add-eval-step-limit-31561' of https://github.com/soghomon-b/transformers into add-eval-step-limit-31561

soghomon-b committed:
ci: retrigger PR slow CI sha detection

soghomon-b commented:
Hi @SunMarc  — apologies for the delay. I’ve just rebased on the latest main, ran make style, and pushed the updated branch.

However, the PR slow CI / Get PR commit SHA better job failed almost immediately. From the logs, it seems like one of the known flakes in pr_run_slow_ci.yml related to commit SHA resolution — could you confirm?

Also, I’m seeing a few test failures in areas I didn’t touch. Could you advise if those are unrelated or if I should look into them?

Let me know if I should re-trigger anything or make further changes — thanks for your time!

SunMarc commented:
> Hi @SunMarc — apologies for the delay. I’ve just rebased on the latest main, ran make style, and pushed the updated branch.
> 
> However, the PR slow CI / Get PR commit SHA better job failed almost immediately. From the logs, it seems like one of the known flakes in pr_run_slow_ci.yml related to commit SHA resolution — could you confirm?
> 
> Also, I’m seeing a few test failures in areas I didn’t touch. Could you advise if those are unrelated or if I should look into them?
> 
> Let me know if I should re-trigger anything or make further changes — thanks for your time!

maybe close this PR then reopen a new one
